### PR TITLE
Ewm5730 diagnostic workspace crash

### PR DESCRIPTION
**Pull request by walshmm**
## Description of work

This pr covers 2 bug fixes:
1. The GroupDiffraction algo trying to join table workspaces with mismatching columns causing it to crash.
2. ~Reduction Workflow unable to proceed because the runnumber submitted to the ReductionRequest object is not a str~  This will be taken care of in pr #393 

## To test

1. Refer to the story for the diffraction calibration input required
2. ~Submit a reduction via the workflow, observe no validation exception occurs.  Ideally reduction should complete granted you have a complete state folder.~

## Link to EWM item

[EWM#5730](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=5730)

### CODE REVIEW

File: src/snapred/backend/recipe/algorithm/BufferMissingColumnsAlgo.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from mantid.api import AlgorithmFactory, ITableWorkspaceProperty, PropertyMode, PythonAlgorithm
+from mantid.kernel import Direction
+
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+
+class BufferMissingColumnsAlgo(PythonAlgorithm):
+    """
+    Record a workspace in a state for the CIS to view later
```
Comment (walshmm) [inline src/snapred/backend/recipe/algorithm/BufferMissingColumnsAlgo.py:9]: I need to still clean this up and write tests for it despite codecov not complaining.

**Comment by rboston628** [comment]
## Review Log for Defect 5730
This is looking at [Defect 5730](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=5730)

### Reproduce  on `next`
On branch `next` on analysis, load:
- run 58882
- lite mode
- group "Column"
- sample = silicon
At tweak peak peek, set:
- FWHM left/right = 4
- Max $chi^2$ = 150

There are now visibly different numbers of peaks in the groups.  Some 3, 4, or 5.
image

Click continue, continue anyway.  Gears whir and spokes turn, and then this:
image

SNAPRed crashes with core dump when warning box is closed.

### Run on this branch
Switch to this branch, and setup exactly as before.  Same peaks result
image

Click continue, continue anyway.

Success!
image

**Comment by rboston628** [comment]
## Review Log for Reduction fix

I'm not sure exactly what I'm supposed to be looking for here.

1. Entered run 46680.  And entering a run is literally the only thing I  can do on this tab.
2. Clicked continue.
3. This error message:
image

File: src/snapred/backend/dao/request/ReductionRequest.py [hunk 2]
```diff
@@ -2,14 +2,12 @@
...
 from pydantic import BaseModel, ConfigDict
 
-from snapred.backend.dao.state.FocusGroup import FocusGroup
 from snapred.backend.error.ContinueWarning import ContinueWarning
 
 
 class ReductionRequest(BaseModel):
     runNumber: Union[str, List[str]]
     useLiteMode: bool
-    focusGroup: Union[Optional[FocusGroup], List[FocusGroup]]
```
Comment (rboston628) [inline src/snapred/backend/dao/request/ReductionRequest.py:12]: I think this is what caused all the failing tests.  I think the issue could be handled on the front-end, by filling in `focusGroup` from the grouping map instead.  It's already populated in the drop down.

**Comment by rboston628** [comment]
Re-ran again as before.  The dspacing table in diagnostic group looks like this:
image
It no longer has the empty rows, and has a row for each detector idea, and a column for all five peaks, even for detectors in a group with 3 or 4 peaks.

The fit params table looks good, and has the peaks labeled by wsindex and peak index
image

Saving completes, and all workspaces are cleared.

**Review by rboston628 — APPROVED** [review]

**Pull request merged**